pkg/util/secretutil/secret.go

```diff
@@ -84,6 +102,19 @@ func getPrivateKey(data []byte) ([]byte, error) {
 		data = rest
 	}
 
+	// if both der and pemBlock are nil, then it might be a pfx cert
+	if der == nil && pemBlock == nil {
```

Comment:
The `for` loop only breaks when `pemBlock == nil` is true. Is the `pemBlock == nil` check here redundant? If yes, can we remove it?